Libplanet.Store/BaseStore.cs

```diff
@@ -46,23 +45,23 @@ public abstract class BaseStore : IStore
             BlockHash branchpoint
         );
 
-        public abstract Transaction GetTransaction(TxId txid);
+        public abstract Transaction? GetTransaction(TxId txid);
 
         public abstract void PutTransaction(Transaction tx);
 
         /// <inheritdoc/>
         public abstract IEnumerable<BlockHash> IterateBlockHashes();
 
         /// <inheritdoc/>
-        public Block GetBlock(BlockHash blockHash)
+        public Block? GetBlock(BlockHash blockHash)
         {
             if (GetBlockDigest(blockHash) is BlockDigest blockDigest)
             {
                 BlockHeader header = blockDigest.GetHeader();
                 (Types.Tx.TxId TxId, Transaction Tx)[] txs = blockDigest.TxIds
                     .Select(bytes => new Types.Tx.TxId(bytes.ToArray()))
                     .OrderBy(txid => txid)
-                    .Select(txid => (txid, GetTransaction(txid)))
+                    .Select(txid => (txid, GetTransaction(txid)!))
```

Comment:
Should change to `(TxId, Transaction?)` instead of using `!`. See line 68.

Reply:
You're right.
But '!' still needs to be used on the line 72.

Reply:
Perhaps I'm not getting it. I don't see why `!` should be used in line 72? You mean in the `return` statement? In that case you can probably restructure the code to have the exact same behavior. 😗

One solution I can think of is to use `(TxId, Transaction)` with only non-`null`s and compare it to the length of `blockDigest` then throw an `Exception` if they don't match. Since an `Exception` shouldn't really be thrown, we can construct a list of missing `TxId`s when creating an `Exception`. This would more likely be more efficient as well. 🙃

Reply:
I tried to solve the problem with minimal code changes. Since the code is designed to throw exceptions explicitly, I will try to restructure it without using the '!' in a way that fits the context.